### PR TITLE
[21783] Update roadmap, release support and versions for `v3.1.0` release

### PR DESCRIPTION
**Pull request by Mario-DL**
## Description

This PR updates the roadmap, release support and versions.md for preparing for `v3.1.0` release.

## Contributor Checklist

- [X] Commit messages follow the project guidelines. 
- **N/A** The code follows the style guidelines of this project. 
- **N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- **N/A** Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- **N/A** Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- **N/A** Changes are API compatible. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 
    
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- **N/A** If this is a critical bug fix, backports to the critical-only supported branches have been requested.
- **N/A** Check CI results: changes do not issue any warning.
- **N/A** Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: versions.md [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+
+Version v3.1.0
+--------------
+
 * Allow `PERSISTENT_DURABILITY` behaving as `TRANSIENT_DURABILITY`. Fallback to `TRANSIENT_LOCAL_DURABILITY` if no persistence guid is set.
 * Fix DomainParticipantQos equality operator by using the new `DomainParticipantQos::compare_flow_controllers`.
+* Added new xml qos overloads for domainparticipant, datawriter and datareader:
+  * ``get_X_qos_from_xml``
+  * ``get_X_qos_from_xml``
+  * ``get_default_X_qos_from_xml``
```
Comment (JesusPoderoso) [inline versions.md:13]: I would add here the dyn net interfaces and remove it from the roadmap, together with a NIT:
```suggestion
* Complete support for dynamic network interfaces
* Add new XML QoS overloads for ``DomainParticipant``, ``DataWriter`` and ``DataReader``:
  * ``get_X_qos_from_xml`` (without profile name)
  * ``get_X_qos_from_xml`` (profile name given)
  * ``get_default_X_qos_from_xml``
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
Just a NIT

File: RELEASE_SUPPORT.md [hunk 2]
```diff
@@ -19,7 +19,9 @@ This period applies since the end of standard support date to the end of life (E
 
 |Version|Version branch|Latest Release|Release Date|End of Standard Support Date|EOL Date|
 |-------|--------------|--------------|------------|----------------------------|--------|
-|3.0|[3.0.x](https://github.com/eProsima/Fast-DDS/tree/3.0.x) (LTS)|[v3.0.0](https://github.com/eProsima/Fast-DDS/releases/tag/v3.0.0)|August 2024|August 2025 [^*]|August 2025 [^*]|
+|3.2|[3.2.x](https://github.com/eProsima/Fast-DDS/tree/3.2.x) (LTS)|-|March 2025|March 2026 [^*]|March 2026 [^*]|
```
Comment (MiguelCompany) [inline RELEASE_SUPPORT.md:22]: I would not put a link here, since the branch does not exist yet

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — APPROVED** [review]

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Pull request merged**